client/extensions/woocommerce/app/store-stats/referrers/index.js

```diff
@@ -21,61 +21,165 @@ import StoreStatsPeriodNav from 'woocommerce/app/store-stats/store-stats-period-
 import JetpackColophon from 'components/jetpack-colophon';
 import Main from 'components/main';
 import Module from 'woocommerce/app/store-stats/store-stats-module';
-import { sortBySales } from './helpers';
+import SearchCard from 'components/search-card';
+import StoreStatsReferrerWidget from 'woocommerce/app/store-stats/store-stats-referrer-widget';
+import { sortBySales } from 'woocommerce/app/store-stats/referrers/helpers';
 
 const STAT_TYPE = 'statsStoreReferrers';
+const LIMIT = 10;
 
-const Referrers = ( { siteId, query, data, selectedDate, unit, slug, translate, queryParams } ) => {
-	const unitSelectedDate = getUnitPeriod( selectedDate, unit );
-	const selectedData = find( data, d => d.date === unitSelectedDate ) || { data: [] };
-	const sortedData = sortBySales( selectedData.data );
-	return (
-		<Main wideLayout>
-			{ siteId && <QuerySiteStats statType={ STAT_TYPE } siteId={ siteId } query={ query } /> }
-			<StoreStatsPeriodNav
-				type="referrers"
-				selectedDate={ selectedDate }
-				unit={ unit }
-				slug={ slug }
-				query={ query }
-				statType={ STAT_TYPE }
-				title={ translate( 'Store Referrers' ) }
-				queryParams={ queryParams }
-			/>
-			<Module
-				siteId={ siteId }
-				emptyMessage={ translate( 'No referrers found' ) }
-				query={ query }
-				statType={ STAT_TYPE }
-			>
-				<table>
-					<tbody>
-						{ sortedData.map( d => (
-							<tr key={ d.referrer }>
-								<td>{ d.date }</td>
-								<td>{ d.referrer }</td>
-								<td>{ d.product_views }</td>
-								<td>{ d.add_to_carts }</td>
-								<td>{ d.product_purchases }</td>
-								<td>${ d.sales }</td>
-							</tr>
-						) ) }
-					</tbody>
-				</table>
-			</Module>
-			<JetpackColophon />
-		</Main>
-	);
-};
+class Referrers extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		query: PropTypes.object.isRequired,
+		data: PropTypes.array.isRequired,
+		selectedDate: PropTypes.string,
+		unit: PropTypes.oneOf( [ 'day', 'week', 'month', 'year' ] ),
+		slug: PropTypes.string,
+		limit: PropTypes.number,
+	};
+
+	state = {
+		filter: '',
+	};
+
+	onSearch = str => {
+		this.setData( this.props, str );
```

Comment:
I agree with @justinshreve's comment that perhaps we should debounce this `onSearch` to prevent setting state multiple times while the search input is keyed in by users.

Reply:
Debounce is useful when you'd like to delay a request when the user is typing. Here, we have the data already. If the purpose of the debounce is to hide the delays in rendering elements, we'd still have to have a spinner or something to let the user know something is happening while we hold off on the computation, filtering and rendering. 

This feels wrong to me because we'd be introducing a delay in order to mask a perfermance issue. I'd rather address the issues itself